File: src/src/CommandBase.cpp
#include "CommandBase.h"

#include <Commands/Scheduler.h>

#include "Subsystems/Drivetrain.h"

// Initialize a single static instance of all of your subsystems. The following
// line should be repeated for each subsystem in the project.
std::unique_ptr<Drivetrain> CommandBase::drivetrain=NULL;
std::unique_ptr<Shooter> CommandBase::shooter = NULL;
std::unique_ptr<OI> CommandBase::oi = NULL;

CommandBase::CommandBase(const std::string &name) : frc::Command(name){

}

void CommandBase::init(){
	CommandBase::drivetrain = std::make_unique<Drivetrain>();
	CommandBase::oi = std::make_unique<OI>();
	CommandBase::shooter = std::make_unique<Shooter>();
}

File: src/org/usfirst/frc/team9514/robot/CommandBase.java
package org.usfirst.frc.team9514.robot;

import org.usfirst.frc.team9514.robot.OI;
import org.usfirst.frc.team9514.robot.subsystems.Climber;
import org.usfirst.frc.team9514.robot.subsystems.Drivetrain;
import org.usfirst.frc.team9514.robot.subsystems.Shooter;
import org.usfirst.frc.team9514.robot.subsystems.Feeder;
import org.usfirst.frc.team9514.robot.subsystems.ExtendingIntake;
import org.usfirst.frc.team9514.robot.subsystems.Gear;

public class CommandBase {
	//Initialize subsystems
	public static Shooter shooter;
	public static Drivetrain drivetrain;
	public static Climber climber;
	public static Feeder feeder;
    public static ExtendingIntake intaker;	
	public static OI oi;
	public static Gear gear;
	
	public CommandBase(){
		
	}
	
	public static void init(){
		oi = new OI();
	    intaker = new ExtendingIntake();	
		shooter = new Shooter();
		drivetrain = new Drivetrain();
		climber = new Climber();
		feeder = new Feeder();
		gear = new Gear();
	}
	
	public static void enable(){
		
	}
	public static void disable(){
		
	}
}
